From 55a1286a6c6170c0fd6fc92997caa7a1032df144 Mon Sep 17 00:00:00 2001
From: Walter Doekes <walter+github@wjd.nu>
Date: Thu, 17 Sep 2020 09:06:09 +0200
Subject: [PATCH] Fix version.h include in release tar (version.h should be in
 include/)

version.h will now be searched first in the build dir, then in the
include. If it's only found in the latter, that one is used.

(When buidling to a directory other than project root, like debhelper
does, the version.h was previously not found because the it searches the
build dir, but not the project root.)

This also reinstates the stub version.h that #errors when you have
downloaded a git snapshot.
---
 CMakeLists.txt                        | 9 ++++++---
 README.md                             | 2 +-
 build.sh                              | 1 +
 include/{stub-version.h => version.h} | 0
 4 files changed, 8 insertions(+), 4 deletions(-)
 rename include/{stub-version.h => version.h} (100%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c4257ef..111c1378 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,11 +11,14 @@ if(EXISTS ./gmock/CMakeLists.txt)
   add_subdirectory(gmock EXCLUDE_FROM_ALL)
 endif()
 
+# Include binary dir first, where we add generated config.h and
+# version.h. If nothing is found there (release tar) use the version.h
+# from the source.
 include_directories(
+  ${PROJECT_SOURCE_DIR}
   ${PROJECT_SOURCE_DIR}/include
   ${PROJECT_SOURCE_DIR}/gtest/include
   ${PROJECT_SOURCE_DIR}/gmock/include
-  ${PROJECT_BINARY_DIR}
   "/usr/local/include")
 
 option(BUILD_STATIC "Build a statically-linked binary" OFF)
@@ -104,8 +107,8 @@ add_custom_target(
     version
     if test -d .git \; then
         ${CMAKE_COMMAND} -D SRC=${PROJECT_SOURCE_DIR}/include/version.h.in
-          -D DST=${PROJECT_BINARY_DIR}/version.h
-          -P ${PROJECT_SOURCE_DIR}/include/version.cmake \;
+        -D DST=${PROJECT_BINARY_DIR}/version.h
+        -P ${PROJECT_SOURCE_DIR}/include/version.cmake \;
     fi
 )
 add_dependencies(sipp version)
diff --git a/README.md b/README.md
index a0001e94..6b201963 100644
--- a/README.md
+++ b/README.md
@@ -66,7 +66,7 @@ list](https://lists.sourceforge.net/lists/listinfo/sipp-users).
 * Update CHANGES.md. Tag release.
 * Make `sipp.1` by calling `help2man --output=sipp.1 -v -v --no-info
   --name='SIP testing tool and traffic generator' ./sipp`
-* Copy `sipp.1`, copy `version.h`.
+* Copy `sipp.1`, copy `$bindir/version.h` to `include/version.h`.
 * Create sipp-VERSION.tar.gz with subdirectory sipp-VERSION. Upload to github as "binary".
 * Run `sudo docker build -t sipp-build docker && sudo docker run -it -v $PWD:/src sipp-build` to create a static binary. Upload this to Github as well.
 
diff --git a/build.sh b/build.sh
index 56a25cb9..a916e441 100755
--- a/build.sh
+++ b/build.sh
@@ -32,4 +32,5 @@ if test -e gtest/.git; then
 	./sipp_unittest
 fi
 
+# You want verbose or NOISY_BUILD? Use VERBOSE=1
 "$MAKE" $MAKEFLAGS
diff --git a/include/stub-version.h b/include/version.h
similarity index 100%
rename from include/stub-version.h
rename to include/version.h
